lib/ThreeScaleClient.php

```diff
@@ -279,11 +339,55 @@ public function authrep($appId, $appKey = null, $usage = null, $userId = null, $
     
   }
 
-  public function authrep_with_user_key($userKey, $usage = null, $userId = null, $object = null, $no_body = null, $serviceId = null) {  
+  /**
+   * Authorize and report in a single shot for user key auth mode
+   *
+   * @param $userKey  User key.
+   * @param $serviceId service id.
+   *
+   * @return ThreeScaleResponse object containing additional authorization information.
+   * If both provider key and application id are valid, the returned object is actually
+   * @see ThreeScaleAuthorizeResponse (which is derived from ThreeScaleResponse) and
+   * contains additional information about the usage status.
+   *
+   * @see ThreeScaleResponse
+   * @see ThreeScaleAuthorizeResponse
+   *
+   * @throws ThreeScaleServerError in case of unexpected internal server error
+   *
+   * Example:
+   *
+   * <code>
+   *   <?php
+   *   $response = $client->authorize('user_key');
+   *   // or $response = $client->authorize('user_key','service_id');
+   *
+   *   if ($response->isSuccess()) {
+   *     // ok.
+   *   } else {
+   *     // something is wrong.
+   *   }
+   *   ?>
+   * </code>
+   */
+
+  public function authrep_with_user_key($userKey, $serviceId = null, $usage = null, $userId = null, $object = null, $no_body = null) {  
```

Comment:
I'm not sure how you called the method before that it did not work. But if someone called it like `authrep_with_user_key('somekey', 15)` where `15` was the `usage` it will become the `serviceId`. That is the breaking change that can break everyone's usage after upgrade.